src/mount.ts

```diff
@@ -41,14 +45,42 @@ interface MountingOptions {
   stubs?: Record<string, any>
 }
 
-export function mount<T extends any>(
-  originalComponent: any,
-  options?: MountingOptions
+// Component declared with defineComponent
+export function mount<
+  TestedComponent extends ComponentPublicInstance,
+  PublicProps extends TestedComponent['$props']
+>(
+  originalComponent: new () => TestedComponent,
+  options?: MountingOptions<PublicProps>
+): VueWrapper<TestedComponent>
+// Component declared with { props: { ... } }
+export function mount<
+  TestedComponent extends ComponentOptionsWithProps,
+  PublicProps extends ExtractPropTypes<TestedComponent['props']>
+>(
+  originalComponent: TestedComponent,
+  options?: MountingOptions<PublicProps>
+): VueWrapper<any>
+// Component declared with { props: [] }
+export function mount<
+  TestedComponent extends ComponentOptionsWithArrayProps,
```

Comment:
This is so ugly lol. I can't believe there isn't a better way without making 1/2 the file definitions.

Reply:
I agree ^^ I spent a long time to make this work, but I think there is no magic possible here: I had to do what Vue itself does... Maybe someone with a better TS knowledge will come with a better solution, but that should still be an improvement in the meantime 🙂 Contributors can play around with it easily: as long as the dts tests are OK, we should be good.

Reply:
cool. I wonder if we can at least hide the types in another file?